include/kernels/coarse_op_preconditioned.cuh

```diff
@@ -1,31 +1,35 @@
+#pragma once
+
 #include <gauge_field_order.h>
 #include <index_helper.cuh>
 #include <matrix_tile.cuh>
 
 namespace quda {
 
-  template <typename Float_, typename PreconditionedGauge, typename Gauge, int n_, int M_, int N_> struct CalculateYhatArg {
+  template <typename Float_, typename PreconditionedGauge, typename Gauge, typename GaugeInv, int n_, int M_, int N_>
```

Comment:
Question for my own reference---is the new template type because `GaugeInv` may have a different data order?

Reply:
Yes it's for the change here: https://github.com/lattice/quda/pull/1006/files#diff-4bc798888295589c7a3032845805baccR269 (not sure if the link works ...). Basically the `Xinv` field is generated with single precision (even if the underlying precision is half) and MILC order, and we can simply use it for the MMA code path, thus the order for `Y` and `Xinv` can be different.